### PR TITLE
[qt6] Make QgsDxfExport Qt6 ready

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: src/core/dxf/qgsdxfexport.h [hunk 1]
```diff
@@ -332,6 +332,14 @@ class CORE_EXPORT QgsDxfExport : public QgsLabelSink
      */
     void writeGroup( int code, int i ) SIP_PYNAME( writeGroupInt );
 
+    /**
+     * Write a tuple of group code and long value
+     * \param code group code
+     * \param i integer value
+     * \note available in Python bindings as writeGroupInt
```
Comment (nyalldawson) [inline src/core/dxf/qgsdxfexport.h:339]:
```suggestion
     * \note available in Python bindings as writeGroupLong
```

File: src/core/dxf/qgsdxfexport.cpp [hunk 2]
```diff
@@ -190,7 +196,11 @@ QgsDxfExport::ExportResult QgsDxfExport::writeToFile( QIODevice *d, const QStrin
   }
 
   mTextStream.setDevice( d );
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
   mTextStream.setCodec( encoding.toLocal8Bit() );
+#else
+  mTextStream.setEncoding( QStringConverter::encodingForName( encoding.toLocal8Bit() ).value_or( QStringConverter::Utf8 ) );
```
Comment (nyalldawson) [inline src/core/dxf/qgsdxfexport.cpp:202]: Cool, I didn't realise qt was using optional...

Comment (m-kuhn) [reply src/core/dxf/qgsdxfexport.cpp:202]: Not an hour between your and my discovery of this

File: python/core/auto_generated/dxf/qgsdxfexport.sip.in [hunk 3]
```diff
@@ -302,6 +302,18 @@ Write a tuple of group code and integer value
 :param code: group code
 :param i: integer value
 
+.. note::
+
+   available in Python bindings as writeGroupInt
```
Comment (m-kuhn) [inline python/core/auto_generated/dxf/qgsdxfexport.sip.in:307]:
```suggestion
   available in Python bindings as writeGroupLong
```

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**